api/app/src/routes/settings.ts

```diff
@@ -46,7 +56,10 @@ router.get(
     if (!settings) {
       settings = await createSettings({ id });
     }
-    return res.status(status.OK).send(SettingsDTO.fromEntity(settings));
+    const { processing, failure } = await countFailedAndProcessingRequests(id);
+    return res
+      .status(status.OK)
+      .send(SettingsDTO.fromEntity(settings, processing, failure));
```

Comment:
General comment on the design:

Coupling the count of webhook requests to the settings get/post seems questionable. Imo getting/updating settings should do exactly that, and counting outstanding webhook stuff should be handled separately.

Otherwise, we'll have to GET settings if we want to access this anywhere else in the dashboard (ie a dedicated webhooks page with a table of reqs), and POST settings will have this overhead attached to it.

Should we consider just adding a "Refresh" button so the user can get the latest count of outstanding requests instead of bundling it with settings?

Reply:
I thought about a refresh button, but didn't want to spend too much time on this, since the same result can be achieved by reloading the page.

As for returning this info with the same DTO, there are pros and cons on both approaches, I went it the simplest for now.

Re: return webhook info with settings
PROS
- simpler
- one request to serve the UI
CONS
- can't get settings without getting webhook info

IMO that's what DTOs are for, they are an independent abstraction from the domain ones, with the goal to better serve a use case/UI. If we ever need to retrieve only WH info, we can build a new endpoint.

Reply:
We discussed offline, its important to build these endpoints more atomic, as there's no association with a UI.

I'll go ahead and split the webhook status from the settings.